protocol/bc/spend.go

```diff
@@ -3,7 +3,7 @@ package bc
 // SpendInput satisfies the TypedInput interface and represents a spend transaction.
 type SpendInput struct {
 	// Commitment
-	Outpoint
+	SpentOutputID OutputID
```

Comment:
Should I also rename `SpentOutputID()` method on TxInput?

Right now it looks like this:

```
func (t *TxInput) SpentOutputID() (o OutputID) {
	if si, ok := t.TypedInput.(*SpendInput); ok {
		o = si.SpentOutputID
	}
	return o
}
```

To me it seems, at the point of call, you never have the actual stutter, but it makes things clearer that the output we are talking about is not in this transaction. Callsites look like:

```
in.SpentOutputID()
inp.SpentOutputID()
txin.SpentOutputID()
etc.
```